### PR TITLE
GH-43119: [CI][Packaging] Update manylinux 2014 CentOS repos that have been deprecated

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Jobs are failing to find mirrorlist.centos.org

### What changes are included in this PR?

Updating repos based on solution from: https://github.com/apache/arrow/issues/43119#issuecomment-2203534492

### Are these changes tested?

Via archery

### Are there any user-facing changes?
No
* GitHub Issue: #43119

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-manylinux-2014*

**Comment by raulcd** [comment]
Testing if the above works. If it does I'll have to update also the java-jni jobs and the r-binary jobs

**Comment by raulcd** [comment]
@github-actions crossbow submit wheel-manylinux-2014-cp310-arm64

**Comment by raulcd** [comment]
It seems that vault.centos.org does not have aarch64, error:
```
failed to solve: process "/bin/sh -c yum install -y dnf" did not complete successfully: exit code: 1
#8 1.314 http://vault.centos.org/centos/7/sclo/aarch64/rh/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
#8 1.314 Trying other mirror.
```
See: https://vault.centos.org/centos/7/sclo/

**Comment by ianmcook** [comment]
@raulcd looks like the aarch64 packages are in a different place: https://vault.centos.org/altarch/

File: ci/docker/python-wheel-manylinux.dockerfile [hunk 1]
```diff
@@ -25,6 +25,11 @@ ARG manylinux
 ENV MANYLINUX_VERSION=${manylinux}
 
 # Ensure dnf is installed, especially for the manylinux2014 base
+RUN if [ "${MANYLINUX_VERSION}" == "2014" ]; then \
+        sed -i s/mirror.centos.org/vault.centos.org/g /etc/yum.repos.d/*.repo && \
+        sed -i s/^#.*baseurl=http/baseurl=http/g /etc/yum.repos.d/*.repo && \
+        sed -i s/^mirrorlist=http/#mirrorlist=http/g /etc/yum.repos.d/*.repo; \
```
Comment (kou) [inline ci/docker/python-wheel-manylinux.dockerfile:31]: We can use one `sed`: https://github.com/apache/arrow/commit/69e8a78c018da88b60f9eb2b3b45703f81f3c93d

**Comment by kou** [comment]
@github-actions crossbow submit wheel-manylinux-2014*

**Comment by raulcd** [comment]
@github-actions crossbow submit java-jars

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by raulcd** [comment]
@kou maybe we should add my last commit to your PR instead but I think is ok to have it here. From what I can see this Dockerfile is only used on the `r-binary-packages` job

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

File: ci/docker/centos-7-cpp.dockerfile [hunk 2]
```diff
@@ -17,11 +17,24 @@
...
+      -e 's/^#baseurl/baseurl/' \
+      -e 's/mirror\.centos\.org/vault.centos.org/' \
+      /etc/yum.repos.d/*.repo
+
 # devtoolset is required for C++17
 RUN \
   yum install -y \
     centos-release-scl \
     epel-release && \
+    sed -i \
```
Comment (kou) [inline ci/docker/centos-7-cpp.dockerfile:33]:
```suggestion
  sed -i \
```

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

**Comment by raulcd** [comment]
@github-actions crossbow submit r-binary-packages

File: ci/docker/centos-7-cpp.dockerfile [hunk 3]
```diff
@@ -17,11 +17,25 @@
...
+
 # devtoolset is required for C++17
 RUN \
   yum install -y \
     centos-release-scl \
     epel-release && \
+  sed -i \
+    -e 's/^mirrorlist/#mirrorlist/' \
+    -e 's/^#baseurl/baseurl/' \
+    -e 's/^# baseurl/baseurl/' \
```
Comment (raulcd) [inline ci/docker/centos-7-cpp.dockerfile:36]: finding this one took a bunch of tests :P

**Review by raulcd — COMMENTED** [review]
:+1:

**Pull request merged**